src/main/java/view/createPostView.java

```diff
@@ -0,0 +1,209 @@
+                createPostViewModel.setState(currentState);  // Set the updated state back
+            }
+
+            @Override
+            public void insertUpdate(DocumentEvent e) {
+                documentListenerHelper();  // Called when text is inserted
+            }
+
+            @Override
+            public void removeUpdate(DocumentEvent e) {
+                documentListenerHelper();  // Called when text is removed
+            }
+
+            @Override
+            public void changedUpdate(DocumentEvent e) {
+                documentListenerHelper();  // Called for attribute changes (rarely needed for plain text input)
+            }
+        });
+    }
+
+
+    private void addSaveButtonListener() {
+        saveButton.addActionListener(e -> {
+            // Gather input data from the user interface
+            String author = homepageController.fetchUser().getUsername(); // TODO: username not name right?
+            String content = contentField.getText();
+            String attachmentPath = null; // TODO: What do we use here
+            String fileType = "text"; // Assume fileType is determined elsewhere
+            int dislikes = 0; // Default value
+            int likes = 0; // Default value
+            String postTitle = titleField.getText();
+            List moderators = new List(); // TODO: Are these two correct
+            List comments = new List();
+            String category = (String) categoryDropdown.getSelectedItem();
+
+            // Validate required inputs
+            // TODO: Not sure how to test if category is null is it categoryDropdown.getSelectedItem() = null?
+            if (postTitle.isEmpty() || content.isEmpty() || author.isEmpty() ) {
+                JOptionPane.showMessageDialog(this, "Title, content, and author are required!");
+                return;
+            }
+
+            // Pass the input data to the controller
+            createPostController.execute(homepageController.fetchUser().getUserID(), content, attachmentPath, fileType, dislikes, likes, postTitle, (java.util.List<User>) moderators, (java.util.List<Comment>) comments, category);
+            // Provide user feedback and switch to the homepage
+            JOptionPane.showMessageDialog(this, "Post created successfully!");
+            createPostController.switchToHomePageview();
+        });
+    }
+
+
+    private void addCancelButtonListener() {
+        cancelButton.addActionListener(e -> handleCancelAction());
+    }
+
+    private void handleCancelAction() {
+        int option = JOptionPane.showConfirmDialog(this,
+                "Are you sure you want to cancel? All unsaved data will be lost.",
+                "Cancel Post Creation", JOptionPane.YES_NO_OPTION);
+
+        if (option == JOptionPane.YES_OPTION) {
+            // Close the current CreatePostView and show the HomePageView
+            this.setVisible(false);
```

Comment:
Incidentally, this.dispose() sounds like better way to get rid of a panel from memory (instead of just hiding it) but this is probably very nit-picky 😃 . Commenting here for historical reference.

Related [StackOverFlow](https://stackoverflow.com/questions/7814758/java-swing-dispose-vs-setvisiblefalse)